### PR TITLE
Components - number not text

### PR DESCRIPTION
**Pull request by brianteeman**
Numerous options in the components are for numbers but the xml field type is text. This pr changes them to the correct value of number.

This ensures that a user cannot enter a letter and only numbers. There are no b/c issues as only a numerical character could ever have worked and there is no change in the way the data is stored.

_I spotted this when debugging a user's site where they put "five" in the count field instead of "5" and wondered why it didnt work. After this PR that isnt possible_

### CODE REVIEW

**Comment by infograf768** [comment]
Don't we need to do something similar to com_config, i.e.
```
filter="integer"
validate="number"
```
and same in the other PR for modules?

**Comment by brianteeman** [comment]
@infograf768 I thought of that but I realised there were a few potential issues

1. They are not all integers
2. Not sure and it needs to be checked but if you had previously entered 2.5 as the value then it might have worked and changing it to an integer would break it
3. Even if they should all be integer (or some other filter) then thats really beyond the scope here as ALL the type="number" fields would also need to be reviewed as many of them also do not have a filter set or use a different way of validating such as a class - we can look at these pr as a first stage

File: administrator/components/com_finder/config.xml [hunk 1]
```diff
@@ -32,7 +32,7 @@
 
 		<field
 			name="description_length"
-			type="text"
+			type="number"
 			label="COM_FINDER_CONFIG_DESCRIPTION_LENGTH_LABEL"
```
Comment (Quy) [inline administrator/components/com_finder/config.xml]: Remove blank line.

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on fc93e6437732d8ce5b2c5a7c44b27daf68aefa08<br /><br />Code review<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/18199">issues.joomla.org/tracker/joomla-cms/18199</a>.</sub>

**Comment by brianteeman** [comment]
fixed conflicts

**Comment by fancyFranci** [comment]
I have tested this item :white_check_mark: successfully on 46f307b1123270aad15771b812b7174661af86b6<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/18199">issues.joomla.org/tracker/joomla-cms/18199</a>.</sub>

**Comment by ghost** [comment]
@Quy can you please retest?

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 46f307b1123270aad15771b812b7174661af86b6<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/18199">issues.joomla.org/tracker/joomla-cms/18199</a>.</sub>

**Comment by ghost** [comment]
RTC after two successful tests.

**Comment by brianteeman** [comment]
Thanks

**Comment by Quy** [comment]
@brianteeman When the field is readonly, there is an up/down incremental control that should probably not be displayed. Perhaps revert for these fields.
number

**Comment by mbabker** [comment]
That sounds like a bug in our polyfill.  Testing another app I have with number fields, those controls do not come up when the field is marked readonly or disabled (and I even went so far as to turn off the CSS to test against plain browser controls).

**Comment by brianteeman** [comment]
Either way it should be created as a new issue

**Comment by Quy** [comment]
The controls only appear in Firefox, but not in Chrome, and Microsoft Edge on Windows 10 Home.

**Comment by dgrammatiko** [comment]
The problem is this line: 
```php
JHtml::_('script', 'system/html5fallback.js', array('version' => 'auto', 'relative' => true));
```

It needs to use the browser conditional (IE<9). This needs to be patched in every form field layout or any form field (libraries) and also any instances in the templates

like:
```php
JHtml::_('script', 'system/html5fallback.js', array('version' => 'auto', 'relative' => true, 'conditional' => 'lt IE 9 '));
```

**Comment by dgrammatiko** [comment]
@Quy please test https://github.com/joomla/joomla-cms/pull/18415

**Comment by infograf768** [comment]
@mbabker 
See https://github.com/joomla/joomla-cms/pull/20632

Do we really have to use the type `number` when the field is readonly?

**Comment by mbabker** [comment]
Semantically, it is correct that a field which can ONLY have numerical data use the number input type, regardless of whether it is editable or readonly.

**Pull request merged**